In [1]:
#Task 1
#import pandas
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib import rcParams
import seaborn as sns
import numpy as np
import os
import sklearn


##import inforamtion from Kaggle 
CCLoan = pd.read_csv(r'D:\springboard\DataScienceGuidedCapstone\Unit7\DataCapstoneProject2\ccFraudDefault\application_data.csv',header=0)

In [2]:
#Task 2
#preview first 25 row to verify data was imported correctly

CCLoan.head(25)

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0       100002       1         Cash loans           M            N   
1       100003       0         Cash loans           F            N   
2       100004       0    Revolving loans           M            Y   
3       100006       0         Cash loans           F            N   
4       100007       0         Cash loans           M            N   
5       100008       0         Cash loans           M            N   
6       100009       0         Cash loans           F            Y   
7       100010       0         Cash loans           M            Y   
8       100011       0         Cash loans           F            N   
9       100012       0    Revolving loans           M            N   
10      100014       0         Cash loans           F            N   
11      100015       0         Cash loans           F            N   
12      100016       0         Cash loans           F            N   
13      100017       0         Cash loans           M            Y   
14      100018       0         Cash loans           F            N   
15      100019       0         Cash loans           M            Y   
16      100020       0         Cash loans           M            N   
17      100021       0    Revolving loans           F            N   
18      100022       0    Revolving loans           F            N   
19      100023       0         Cash loans           F            N   
20      100024       0    Revolving loans           M            Y   
21      100025       0         Cash loans           F            Y   
22      100026       0         Cash loans           F            N   
23      100027       0         Cash loans           F            N   
24      100029       0         Cash loans           M            Y   

   FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0                Y             0        202500.000    406597.5      24700.5   
1                N             0        270000.000   1293502.5      35698.5   
2                Y             0         67500.000    135000.0       6750.0   
3                Y             0        135000.000    312682.5      29686.5   
4                Y             0        121500.000    513000.0      21865.5   
5                Y             0         99000.000    490495.5      27517.5   
6                Y             1        171000.000   1560726.0      41301.0   
7                Y             0        360000.000   1530000.0      42075.0   
8                Y             0        112500.000   1019610.0      33826.5   
9                Y             0        135000.000    405000.0      20250.0   
10               Y             1        112500.000    652500.0      21177.0   
11               Y             0         38419.155    148365.0      10678.5   
12               Y             0         67500.000     80865.0       5881.5   
13               N             1        225000.000    918468.0      28966.5   
14               Y             0        189000.000    773680.5      32778.0   
15               Y             0        157500.000    299772.0      20160.0   
16               N             0        108000.000    509602.5      26149.5   
17               Y             1         81000.000    270000.0      13500.0   
18               Y             0        112500.000    157500.0       7875.0   
19               Y             1         90000.000    544491.0      17563.5   
20               Y             0        135000.000    427500.0      21375.0   
21               Y             1        202500.000   1132573.5      37561.5   
22               N             1        450000.000    497520.0      32521.5   
23               Y             0         83250.000    239850.0      23850.0   
24               N             2        135000.000    247500.0      12703.5   

    ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20 FLAG_DOCUMENT_21  \
0   ...                 0                0      

In [3]:
#Task 3
#Total row of record is 307511
CCLoan.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 307511 entries, 0 to 307510
Columns: 122 entries, SK_ID_CURR to AMT_REQ_CREDIT_BUREAU_YEAR
dtypes: float64(65), int64(41), object(16)
memory usage: 286.2+ MB


I plan to remove columns that have more than 40% (123004) or more data missing of fill out some of the column that are avg.  

In [4]:
#Task 3.1.0
#see what data is null, total row is 307511 and the below is top 5 row with 214000+ rows are null

aaa=CCLoan.isna().sum()
aaa.sort_values(ascending = False).head(50)


#COMMONAREA_MEDI             214865
#COMMONAREA_AVG              214865
#COMMONAREA_MODE             214865
#NONLIVINGAPARTMENTS_MODE    213514
#NONLIVINGAPARTMENTS_AVG     213514


COMMONAREA_MEDI             214865
COMMONAREA_AVG              214865
COMMONAREA_MODE             214865
NONLIVINGAPARTMENTS_MODE    213514
NONLIVINGAPARTMENTS_AVG     213514
                             ...  
NAME_HOUSING_TYPE                0
NAME_FAMILY_STATUS               0
NAME_EDUCATION_TYPE              0
NAME_INCOME_TYPE                 0
SK_ID_CURR                       0
Length: 122, dtype: int64

In [5]:
#Task 3.1.1.0

#drop column COMMONAREA_MEDI, COMMONAREA_AVG, COMMONAREA_MODE, NONLIVINGAPARTMENTS_MODE, NONLIVINGAPARTMENTS_AVG
CCLoan_clean=CCLoan.drop(['COMMONAREA_MEDI','COMMONAREA_AVG','COMMONAREA_MODE','NONLIVINGAPARTMENTS_MODE','NONLIVINGAPARTMENTS_AVG'],axis=1)


#validate columns are dropped
#validated the columns are no longer there and now see next 5 columns
bbb=CCLoan_clean.isna().sum()
bbb.sort_values(ascending = False,)

NONLIVINGAPARTMENTS_MEDI    213514
FONDKAPREMONT_MODE          210295
LIVINGAPARTMENTS_AVG        210199
LIVINGAPARTMENTS_MODE       210199
LIVINGAPARTMENTS_MEDI       210199
                             ...  
NAME_HOUSING_TYPE                0
NAME_FAMILY_STATUS               0
NAME_EDUCATION_TYPE              0
NAME_INCOME_TYPE                 0
SK_ID_CURR                       0
Length: 117, dtype: int64

In [6]:
#3.1.2.0

#drop column NONLIVINGAPARTMENTS_MEDI, FONDKAPREMONT_MODE, LIVINGAPARTMENTS_AVG, LIVINGAPARTMENTS_MODE, LIVINGAPARTMENTS_MEDI
CCLoan_clean=CCLoan_clean.drop(['NONLIVINGAPARTMENTS_MEDI','FONDKAPREMONT_MODE','LIVINGAPARTMENTS_AVG','LIVINGAPARTMENTS_MODE','LIVINGAPARTMENTS_MEDI'],axis =1)

#validate columns are dropped
#validated the columns are no longer there and now see next 5 columns
ccc=CCLoan_clean.isna().sum()
ccc.sort_values(ascending = False,)

FLOORSMIN_AVG          208642
FLOORSMIN_MODE         208642
FLOORSMIN_MEDI         208642
YEARS_BUILD_AVG        204488
YEARS_BUILD_MODE       204488
                        ...  
NAME_HOUSING_TYPE           0
NAME_FAMILY_STATUS          0
NAME_EDUCATION_TYPE         0
NAME_INCOME_TYPE            0
SK_ID_CURR                  0
Length: 112, dtype: int64

In [7]:
#3.1.3.0

#drop column FLOORSMIN_AVG, FLOORSMIN_MODE, FLOORSMIN_MEDI, YEARS_BUILD_AVG , YEARS_BUILD_MODE, 
CCLoan_clean=CCLoan_clean.drop(['FLOORSMIN_AVG','FLOORSMIN_MODE','FLOORSMIN_MEDI', 'YEARS_BUILD_AVG', 'YEARS_BUILD_MODE'],axis =1)

#validate columns are dropped
ddd=CCLoan_clean.isna().sum()
ddd.sort_values(ascending = False,)

YEARS_BUILD_MEDI       204488
OWN_CAR_AGE            202929
LANDAREA_MEDI          182590
LANDAREA_AVG           182590
LANDAREA_MODE          182590
                        ...  
NAME_HOUSING_TYPE           0
NAME_FAMILY_STATUS          0
NAME_EDUCATION_TYPE         0
NAME_INCOME_TYPE            0
SK_ID_CURR                  0
Length: 107, dtype: int64

In [8]:
#3.1.4.0

#drop column YEARS_BUILD_MEDI ,LANDAREA_MEDI ,LANDAREA_AVG, LANDAREA_MODE  
# i am keeping the own car age as it could be useful to know the age of vehicle.  I notice this is NaN when they dont have a car.
CCLoan_clean=CCLoan_clean.drop(['YEARS_BUILD_MEDI','LANDAREA_MEDI','LANDAREA_AVG','LANDAREA_MODE'],axis =1)

#validate columns are dropped
eee=CCLoan_clean.isna().sum()
eee.sort_values(ascending = False,)

OWN_CAR_AGE            202929
BASEMENTAREA_AVG       179943
BASEMENTAREA_MODE      179943
BASEMENTAREA_MEDI      179943
EXT_SOURCE_1           173378
                        ...  
NAME_HOUSING_TYPE           0
NAME_FAMILY_STATUS          0
NAME_EDUCATION_TYPE         0
NAME_INCOME_TYPE            0
SK_ID_CURR                  0
Length: 103, dtype: int64

In [9]:
#3.1.5.0

#drop column BASEMENTAREA_AVG, BASEMENTAREA_MODE, BASEMENTAREA_MEDI, EXT_SOURCE_1 
# i am keeping the own car age as it could be useful to know the age of vehicle.  I notice this is NaN when they dont have a car.
CCLoan_clean=CCLoan_clean.drop(['BASEMENTAREA_AVG','BASEMENTAREA_MODE','BASEMENTAREA_MEDI','EXT_SOURCE_1'],axis =1)

#validate columns are dropped
aaa=CCLoan_clean.isna().sum()
aaa.sort_values(ascending = False,)

OWN_CAR_AGE            202929
NONLIVINGAREA_AVG      169682
NONLIVINGAREA_MEDI     169682
NONLIVINGAREA_MODE     169682
ELEVATORS_AVG          163891
                        ...  
NAME_HOUSING_TYPE           0
NAME_FAMILY_STATUS          0
NAME_EDUCATION_TYPE         0
NAME_INCOME_TYPE            0
SK_ID_CURR                  0
Length: 99, dtype: int64

In [11]:
#3.1.6.0

#drop column NONLIVINGAREA_AVG, NONLIVINGAREA_MEDI ,NONLIVINGAREA_MODE, ELEVATORS_AVG 
CCLoan_clean=CCLoan_clean.drop(['NONLIVINGAREA_AVG','NONLIVINGAREA_MEDI','NONLIVINGAREA_MODE','ELEVATORS_AVG'],axis =1)

#validate columns are dropped
aaa=CCLoan_clean.isna().sum()
aaa.sort_values(ascending = False)

OWN_CAR_AGE            202929
ELEVATORS_MEDI         163891
ELEVATORS_MODE         163891
WALLSMATERIAL_MODE     156341
APARTMENTS_AVG         156061
                        ...  
NAME_HOUSING_TYPE           0
NAME_FAMILY_STATUS          0
NAME_EDUCATION_TYPE         0
NAME_INCOME_TYPE            0
SK_ID_CURR                  0
Length: 95, dtype: int64

In [13]:
#3.1.7.0

#drop column ELEVATORS_MEDI, ELEVATORS_MODE, WALLSMATERIAL_MODE, APARTMENTS_AVG 
CCLoan_clean=CCLoan_clean.drop(['ELEVATORS_MEDI','ELEVATORS_MODE','WALLSMATERIAL_MODE','APARTMENTS_AVG'],axis =1)

#validate columns are dropped
aaa=CCLoan_clean.isna().sum()
aaa.sort_values(ascending = False,)

OWN_CAR_AGE            202929
APARTMENTS_MEDI        156061
APARTMENTS_MODE        156061
ENTRANCES_MEDI         154828
ENTRANCES_AVG          154828
                        ...  
NAME_HOUSING_TYPE           0
NAME_FAMILY_STATUS          0
NAME_EDUCATION_TYPE         0
NAME_INCOME_TYPE            0
SK_ID_CURR                  0
Length: 91, dtype: int64

In [14]:
#3.1.8.0

#drop column APARTMENTS_MEDI, APARTMENTS_MODE, ENTRANCES_MEDI, ENTRANCES_AVG 
CCLoan_clean=CCLoan_clean.drop(['APARTMENTS_MEDI', 'APARTMENTS_MODE', 'ENTRANCES_MEDI', 'ENTRANCES_AVG'],axis =1)

#validate columns are dropped
aaa=CCLoan_clean.isna().sum()
aaa.sort_values(ascending = False,)


OWN_CAR_AGE            202929
ENTRANCES_MODE         154828
LIVINGAREA_MEDI        154350
LIVINGAREA_AVG         154350
LIVINGAREA_MODE        154350
                        ...  
NAME_HOUSING_TYPE           0
NAME_FAMILY_STATUS          0
NAME_EDUCATION_TYPE         0
NAME_INCOME_TYPE            0
SK_ID_CURR                  0
Length: 87, dtype: int64

In [15]:
#3.1.9.0

#drop column ENTRANCES_MODE, LIVINGAREA_MEDI, LIVINGAREA_AVG, APARTMENTS_AVG 
CCLoan_clean=CCLoan_clean.drop(['ENTRANCES_MODE', 'LIVINGAREA_MEDI', 'LIVINGAREA_AVG', 'LIVINGAREA_MODE'],axis =1)

#validate columns are dropped
aaa=CCLoan_clean.isna().sum()
aaa.sort_values(ascending = False,)

OWN_CAR_AGE            202929
HOUSETYPE_MODE         154297
FLOORSMAX_AVG          153020
FLOORSMAX_MODE         153020
FLOORSMAX_MEDI         153020
                        ...  
NAME_HOUSING_TYPE           0
NAME_FAMILY_STATUS          0
NAME_EDUCATION_TYPE         0
NAME_INCOME_TYPE            0
SK_ID_CURR                  0
Length: 83, dtype: int64

In [16]:
#3.1.10.0

#drop column HOUSETYPE_MODE, FLOORSMAX_AVG, FLOORSMAX_MODE, FLOORSMAX_MEDI 
CCLoan_clean=CCLoan_clean.drop(['HOUSETYPE_MODE', 'FLOORSMAX_AVG', 'FLOORSMAX_MODE', 'FLOORSMAX_MEDI'],axis =1)

#validate columns are dropped
aaa=CCLoan_clean.isna().sum()
aaa.sort_values(ascending = False,)


OWN_CAR_AGE                     202929
YEARS_BEGINEXPLUATATION_AVG     150007
YEARS_BEGINEXPLUATATION_MEDI    150007
YEARS_BEGINEXPLUATATION_MODE    150007
TOTALAREA_MODE                  148431
                                 ...  
REG_REGION_NOT_WORK_REGION           0
LIVE_REGION_NOT_WORK_REGION          0
REG_CITY_NOT_LIVE_CITY               0
REG_CITY_NOT_WORK_CITY               0
LIVE_CITY_NOT_WORK_CITY              0
Length: 79, dtype: int64

In [17]:
#3.1.11.0

#drop column YEARS_BEGINEXPLUATATION_AVG, YEARS_BEGINEXPLUATATION_MEDI, YEARS_BEGINEXPLUATATION_MODE, TOTALAREA_MODE 
CCLoan_clean=CCLoan_clean.drop(['YEARS_BEGINEXPLUATATION_AVG', 'YEARS_BEGINEXPLUATATION_MEDI', 'YEARS_BEGINEXPLUATATION_MODE', 'TOTALAREA_MODE'],axis =1)

#validate columns are dropped
aaa=CCLoan_clean.isna().sum()
aaa.sort_values(ascending = False,)

#

OWN_CAR_AGE                    202929
EMERGENCYSTATE_MODE            145755
OCCUPATION_TYPE                 96391
EXT_SOURCE_3                    60965
AMT_REQ_CREDIT_BUREAU_HOUR      41519
                                ...  
REG_REGION_NOT_WORK_REGION          0
LIVE_REGION_NOT_WORK_REGION         0
TARGET                              0
REG_CITY_NOT_WORK_CITY              0
REG_CITY_NOT_LIVE_CITY              0
Length: 75, dtype: int64

In [ ]:
#3.1.12.0
#dropped all columns that have atleast more than 40% data available
#drop column YEARS_BEGINEXPLUATATION_AVG, YEARS_BEGINEXPLUATATION_MEDI, YEARS_BEGINEXPLUATATION_MODE, TOTALAREA_MODE 
#CCLoan_clean=CCLoan_clean.drop(['YEARS_BEGINEXPLUATATION_AVG', 'YEARS_BEGINEXPLUATATION_MEDI', 'YEARS_BEGINEXPLUATATION_MODE', 'TOTALAREA_MODE'],axis =1)

#validate columns are dropped
#aaa=CCLoan_clean.isna().sum()
#aaa.sort_values(ascending = False,)

In [19]:
CCLoan_clean[['EMERGENCYSTATE_MODE','OCCUPATION_TYPE','EXT_SOURCE_3','AMT_REQ_CREDIT_BUREAU_HOUR']].head(25)

EMERGENCYSTATE_MODE OCCUPATION_TYPE  EXT_SOURCE_3  \
0                   No        Laborers      0.139376   
1                   No      Core staff           NaN   
2                  NaN        Laborers      0.729567   
3                  NaN        Laborers           NaN   
4                  NaN      Core staff           NaN   
5                  NaN        Laborers      0.621226   
6                  NaN     Accountants      0.492060   
7                  NaN        Managers      0.540654   
8                  NaN             NaN      0.751724   
9                  NaN        Laborers           NaN   
10                 NaN      Core staff      0.363945   
11                 NaN             NaN      0.652897   
12                  No        Laborers      0.176653   
13                  No         Drivers      0.770087   
14                  No        Laborers           NaN   
15                 NaN        Laborers      0.678568   
16                 NaN         Drivers      0.062103   
17                 NaN        Laborers           NaN   
18                  No        Laborers      0.556727   
19                 NaN      Core staff      0.477649   
20                  No        Laborers           NaN   
21                 NaN     Sales staff      0.542445   
22                  No     Sales staff      0.358951   
23                  No             NaN      0.669057   
24                  No         Drivers      0.565608   

    AMT_REQ_CREDIT_BUREAU_HOUR  
0                          0.0  
1                          0.0  
2                          0.0  
3                          NaN  
4                          0.0  
5                          0.0  
6                          0.0  
7                          0.0  
8                          0.0  
9                          NaN  
10                         0.0  
11                         0.0  
12                         0.0  
13                         0.0  
14                         NaN  
15                         0.0  
16                         0.0  
17                         NaN  
18                         0.0  
19                         0.0  
20                         NaN  
21                         0.0  
22                         0.0  
23                         0.0  
24                         0.0

In [ ]:
CCLoan_clean.info()

In [ ]:
#Task 4
#rename certian column to be more readable
CCLoan.rename(columns={'SK_ID_CURR':'CLIENTID','TARGET':'LATE_PAYMENT'},inplace = True)


In [ ]:
#Task 5
CCLoan.head(25)


In [ ]:
#Task 5.1
CCLoan.describe()


In this section 5.2.x i will look at each of the 122 column and see the number null value.  It will be better if most columns are not null and have NaN

In [ ]:
#Task 5.2.1
#Checking first 1-13 column of missing data 

CC_Late = CCLoan['LATE_PAYMENT'].isnull()
print ('Late Payment Missing: '+str(CC_Late.sum()))

CC_CONTRACT = CCLoan['NAME_CONTRACT_TYPE'].isnull()
print ('Contract Type Missing: '+str(CC_CONTRACT.sum()))

CC_GENDER=CCLoan['CODE_GENDER'].isnull()
print ('Gender Missing: '+str(CC_GENDER.sum()))

CC_OWNCAR=CCLoan['FLAG_OWN_CAR'].isnull()
print ('Own Car Missing: '+str(CC_OWNCAR.sum()))

CC_OWNREALITY=CCLoan['FLAG_OWN_REALTY'].isnull()
print ('Own Reality Missing: '+str(CC_OWNREALITY.sum()))

CC_CHILD =CCLoan['CNT_CHILDREN'].isnull()
print ('Own Reality Missing: '+str(CC_OWNREALITY.sum()))

CC_INCOME=CCLoan['AMT_INCOME_TOTAL'].isnull()
print ('Own Reality Missing: '+str(CC_INCOME.sum()))

CC_CREDIT=CCLoan['AMT_CREDIT'].isnull()
print ('Own Reality Missing: '+str(CC_CREDIT.sum()))

CC_ANNUITY=CCLoan['AMT_ANNUITY'].isnull()
print ('Annuity Missing: '+str(CC_ANNUITY.sum()))

CC_GOODSPRICE=CCLoan['AMT_GOODS_PRICE'].isnull()
print ('Goods Price Missing: '+str(CC_GOODSPRICE.sum()))




In [ ]:
#Task 5.2.1.1
#show the 12 AMT_ANNUITY is missing. 
#Looking at data says it is NaN.  So it is fine
CC_ANNUITY1 = CCLoan.loc[CCLoan['AMT_ANNUITY'].isnull()]
CC_ANNUITY1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','AMT_GOODS_PRICE']].head(25)

In [ ]:
#Task 5.2.1.2
#It is not missing only NaN
CC_GOODSPRICE1= CCLoan.loc[CCLoan['AMT_GOODS_PRICE'].isnull()]
CC_GOODSPRICE1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','AMT_GOODS_PRICE']].head(25)
#check if there is zero price(mean price) and price likely 0

In [ ]:
#Task 5.2.2
#Checking next 14-23 column of missing data 

CC_TYPESUIT = CCLoan['NAME_TYPE_SUITE'].isnull()
print ('NAME_TYPE_SUITE: '+str(CC_TYPESUIT.sum()))

CC_INCOMETYPE = CCLoan['NAME_INCOME_TYPE'].isnull()
print ('NAME_INCOME_TYPE Missing: '+str(CC_INCOMETYPE.sum()))

CC_EDUCATIONTYPE=CCLoan['NAME_EDUCATION_TYPE'].isnull()
print ('NAME_EDUCATION_TYPE Missing: '+str(CC_EDUCATIONTYPE.sum()))

CC_FAMILY=CCLoan['NAME_FAMILY_STATUS'].isnull()
print ('NAME_FAMILY_STATUS Missing: '+str(CC_FAMILY.sum()))

CC_HOUSETYPE=CCLoan['NAME_HOUSING_TYPE'].isnull()
print ('NAME_HOUSING_TYPE Missing: '+str(CC_HOUSETYPE.sum()))

CC_REGION_POPULATION =CCLoan['REGION_POPULATION_RELATIVE'].isnull()
print ('REGION_POPULATION_RELATIVE Missing: '+str(CC_REGION_POPULATION.sum()))

CC_DAYSBIRTH=CCLoan['DAYS_BIRTH'].isnull()
print ('DAYS_BIRTH Missing: '+str(CC_DAYSBIRTH.sum()))

CC_EMPLOYED=CCLoan['DAYS_EMPLOYED'].isnull()
print ('DAYS_EMPLOYED Missing: '+str(CC_EMPLOYED.sum()))

CC_ANNUITY=CCLoan['DAYS_REGISTRATION'].isnull()
print ('ANNUITY Missing: '+str(CC_ANNUITY.sum()))

CC_GOODSPRICE=CCLoan['DAYS_ID_PUBLISH'].isnull()
print ('Goods Price Missing: '+str(CC_GOODSPRICE.sum()))

In [ ]:
#Task 5.2.2.1
#It is not missing only NaN because maybe no one came with them to borrow loan
CC_NAMETYPE1= CCLoan.loc[CCLoan['NAME_TYPE_SUITE'].isnull()]
CC_NAMETYPE1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','NAME_TYPE_SUITE']].head(25)
#rename 

In [ ]:
#Task 5.2.3
#Checking next 24-33 column of missing data 

CC_CARAGE = CCLoan['OWN_CAR_AGE'].isnull()
print ('OWN_CAR_AGE Missing: '+str(CC_CARAGE.sum()))

CC_MOBIL = CCLoan['FLAG_MOBIL'].isnull()
print ('CC_MOBIL Missing: '+str(CC_MOBIL.sum()))

CC_EMPPHONE=CCLoan['FLAG_EMP_PHONE'].isnull()
print ('CC_EMPPHONE Missing: '+str(CC_EMPPHONE.sum()))

CC_WORKPHONE=CCLoan['FLAG_WORK_PHONE'].isnull()
print ('CC_WORKPHONE: '+str(CC_WORKPHONE.sum()))

CC_CONTMOBILE=CCLoan['FLAG_CONT_MOBILE'].isnull()
print ('CC_CONTMOBILE Missing: '+str(CC_CONTMOBILE.sum()))

CC_PHONE =CCLoan['FLAG_PHONE'].isnull()
print ('CC_PHONE Missing: '+str(CC_PHONE.sum()))

CC_EMAIL=CCLoan['FLAG_EMAIL'].isnull()
print ('CC_EMAIL Missing: '+str(CC_EMAIL.sum()))

CC_OCCUPATIONTYPE=CCLoan['OCCUPATION_TYPE'].isnull()
print ('CC_OCCUPATIONTYPE Missing: '+str(CC_OCCUPATIONTYPE.sum()))

CC_FAMMEMEBER=CCLoan['CNT_FAM_MEMBERS'].isnull()
print ('CC_FAMMEMEBER Missing: '+str(CC_FAMMEMEBER.sum()))

CC_CLIENTRATING=CCLoan['REGION_RATING_CLIENT'].isnull()
print ('CC_CLIENTRATING Missing: '+str(CC_CLIENTRATING.sum()))

In [ ]:
#Task 5.2.3.1
#It is not missing only NaN because maybe no one came with them to borrow loan
CC_OWN_CAR_AGE1= CCLoan.loc[CCLoan['OWN_CAR_AGE'].isnull()]
CC_OWN_CAR_AGE1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','OWN_CAR_AGE']].head(25)

In [ ]:
#Task 5.2.3.2
#It is not missing only NaN because maybe no one came with them to borrow loan
CC_OCCUPATION_TYPE1= CCLoan.loc[CCLoan['OCCUPATION_TYPE'].isnull()]
CC_OCCUPATION_TYPE1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','OCCUPATION_TYPE']].head(25)


In [ ]:
#Task 5.2.3.3
#It is not missing only NaN because maybe no one came with them to borrow loan
CC_CLIENTRATING1= CCLoan.loc[CCLoan['REGION_RATING_CLIENT'].isnull()]
CC_CLIENTRATING1 [['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','REGION_RATING_CLIENT']].head(25)

In [ ]:

#Task 5.2.4
#Checking next 34-43 column of missing data 

CC_CLIENTCITYRATING = CCLoan['REGION_RATING_CLIENT_W_CITY'].isnull()
print ('CC_CLIENTCITYRATING Missing: '+str(CC_CLIENTCITYRATING.sum()))

CC_WEEKDAY_APPR = CCLoan['WEEKDAY_APPR_PROCESS_START'].isnull()
print ('CC_WEEKDAY_APPR Missing: '+str(CC_WEEKDAY_APPR.sum()))

CC_HOUR_APPR=CCLoan['HOUR_APPR_PROCESS_START'].isnull()
print ('Gender Missing: '+str(CC_HOUR_APPR.sum()))

CC_REGNOTLIVE=CCLoan['REG_REGION_NOT_LIVE_REGION'].isnull()
print ('CC_HOUR_APPR Missing: '+str(CC_REGNOTLIVE.sum()))

CC_REGIONNOTWORK=CCLoan['REG_REGION_NOT_WORK_REGION'].isnull()
print ('CC_REGIONNOTWORK Missing: '+str(CC_OWNREALITY.sum()))

CC_LIVEREGION =CCLoan['LIVE_REGION_NOT_WORK_REGION'].isnull()
print ('CC_LIVEREGION Missing: '+str(CC_OWNREALITY.sum()))

CC_CITYNOTLIVE=CCLoan['REG_CITY_NOT_LIVE_CITY'].isnull()
print ('CC_CITYNOTLIVE Missing: '+str(CC_INCOME.sum()))

CC_CITYNOTWORK=CCLoan['REG_CITY_NOT_WORK_CITY'].isnull()
print ('CC_CITYNOTWORK Missing: '+str(CC_CREDIT.sum()))

CC_LIVECITY=CCLoan['LIVE_CITY_NOT_WORK_CITY'].isnull()
print ('CC_LIVECITY Missing: '+str(CC_ANNUITY.sum()))

CC_GOODSPRICE=CCLoan['ORGANIZATION_TYPE'].isnull()
print ('CC_GOODSPRICE Missing: '+str(CC_GOODSPRICE.sum()))


In [ ]:
#Task 5.2.5
#Checking COLUMN 


CC_EXT_SOURCE_1 = CCLoan['EXT_SOURCE_1'].isnull()
print ('EXT_SOURCE_1 Missing: '+str(CC_Late.sum()))

CC_EXT_SOURCE_2 = CCLoan['EXT_SOURCE_2'].isnull()
print ('EXT_SOURCE_2 Missing: '+str(CC_CONTRACT.sum()))

EXT_SOURCE_3=CCLoan['EXT_SOURCE_3'].isnull()
print ('EXT_SOURCE_3 Missing: '+str(CC_GENDER.sum()))

CC_APARTMENTS_AVG=CCLoan['APARTMENTS_AVG'].isnull()
print ('APARTMENTS_AVG Missing: '+str(CC_OWNCAR.sum()))

CC_BASEMENTAREA_AVG=CCLoan['BASEMENTAREA_AVG'].isnull()
print ('BASEMENTAREA_AVG Missing: '+str(CC_OWNREALITY.sum()))

CC_DAYS_LAST_PHONE_CHANGE=CCLoan['DAYS_LAST_PHONE_CHANGE'].isnull()
print ('DAYS_LAST_PHONE_CHANGE Missing: '+str(CC_DAYS_LAST_PHONE_CHANGE.sum()))

#CC_CHILD =CCLoan['LIVE_REGION_NOT_WORK_REGION'].isnull()
#print ('Own Reality Missing: '+str(CC_OWNREALITY.sum()))

CC_CREDITMONTH=CCLoan['AMT_REQ_CREDIT_BUREAU_MON'].isnull()
print ('Credit Month Missing: '+str(CC_CREDITMONTH.sum()))

CC_CREDITQRT=CCLoan['AMT_REQ_CREDIT_BUREAU_QRT'].isnull()
print ('Credit Quarter Missing: '+str(CC_CREDIT.sum()))

CC_CREDITYEAR=CCLoan['AMT_REQ_CREDIT_BUREAU_YEAR'].isnull()
print ('Credit Year Missing: '+str(CC_CREDITYEAR.sum()))





In [ ]:
CCLoan['FLAG_OWN_CAR'].count()

In [ ]:
LatePay= CCLoan.loc[CCLoan['LATE_PAYMENT']==1]
NoLate = CCLoan.loc[CCLoan['LATE_PAYMENT']==0]

In [ ]:
#percent of people who are late vs nonlate on cc payment
print ('% of Late Payment : ' +str(len(LatePay['CLIENTID'])/len(CCLoan['CLIENTID'])*100))
print ('% in good standing : ' +str(len(NoLate['CLIENTID'])/len(CCLoan['CLIENTID'])*100))


#there is a 8% default on credit


In [ ]:
NoLate

In [ ]:
#Task 6.1
#Find what is percentage who is Late Payment that owns a car

owncar_LatePayment=(len(LatePay[LatePay['FLAG_OWN_CAR']=='Y']))/len(LatePay['FLAG_OWN_CAR'])*100
print ('% whom has defaulted: '+ str(owncar_LatePayment))


owncar_NoLate=(len(NoLate[NoLate['FLAG_OWN_CAR']=='Y']))/len(NoLate['FLAG_OWN_CAR'])*100
print('% that has not defaulted : '+ str(owncar_NoLate))


#there is a 4% difference where person who owns car that didnt default as to to whom has

In [ ]:
Reality
No Reality

In [ ]:
#Task 6.2
#Find what is percentage who is Late Payment that owns reality

owncar_LatePayment=(len(LatePay[LatePay['FLAG_OWN_CAR']=='Y']))/len(LatePay['FLAG_OWN_CAR'])*100
print ('% whom has defaulted: '+ str(owncar_LatePayment))


owncar_NoLate=(len(NoLate[NoLate['FLAG_OWN_CAR']=='Y']))/len(NoLate['FLAG_OWN_CAR'])*100
print('% that has not defaulted : '+ str(owncar_NoLate))



In [ ]:
#
#should do a graph of client ID vs the amount AMT_GOOD_PRICE / AMT_CREDIT

In [ ]:
#
#should do a graph of client ID vs the amount AMT_CREDIT / AMT_INCOME_TOTAL

In [ ]:
#Task x.6.5
#total people with cars and then information if they are late or not late on payment
car_owners = CCLoan[CCLoan['FLAG_OWN_CAR']== 'Y']
car_owner_T = car_owners[car_owners['LATE_PAYMENT']== 1]
car_owner_NT = car_owners[car_owners['LATE_PAYMENT']== 0]
print('Total Count:    '+ str(car_owners['LATE_PAYMENT'].count()))  #.sum() properway
print('LatePayment:    '+ str(car_owner_T['LATE_PAYMENT'].count()))
print('No LatePayment: '+ str(car_owner_NT['LATE_PAYMENT'].count()))

In [ ]:
#Task 6
#Find the % that is late vs not late
#8.1 % of total client is late
VS = (Late['CLIENTID'].count() / CCLoan['CLIENTID'].count())*100

print(VS)


In [ ]:
#Task 7
#get a list of clients who have a late installment
Late = CCLoan.loc[CCLoan['LATE_PAYMENT']. isin([1])]

In [ ]:
#Task 8
#count number of Target are late ---24825 is the output
Late.count()

In [ ]:
#Task 9
#try to find relationship where 
Late.head(25)


In [ ]:
#Task 10
Late[['CLIENTID','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','AMT_GOODS_PRICE']].head(25)

In [ ]:
#Task 11
Late.describe()

In [ ]:
#Task 21
#put non late payment into another DF so I can graph the latepayment vs income
NonLate = CCLoan.loc[CCLoan['LATE_PAYMENT']. isin([0])]

In [ ]:
CCLoan[['CLIENTID','LATE_PAYMENT','AMT_INCOME_TOTAL','AMT_CREDIT','AMT_ANNUITY','AMT_GOODS_PRICE','FLAG_OWN_CAR']].head(25)

#Task 22
#graph amt_credit vs amt_annuity

y = NonLate['AMT_CREDIT']
y2 = NonLate['AMT_ANNUITY']

x = NonLate ['CLIENT']
fig = plt.figure()
ax = plt.subplot()
ax.plot(x, y, label='Credit Amount')
ax.plot(x, y2, label='Loan Payment')
plt.title('Legend inside')
ax.legend()


In [ ]:
#GRAPH target vs points FLAG_OWN_CAR, FLAG_OWN_REALTY




In [ ]:
#Task 1000
#testsplit  dataset
from sklearn.model_selection import train_test_split

train_test_split(x,y,test_size = .2)